### PR TITLE
Issue #5552 - Verify a defined class is resolved `whenDefined`

### PR DESCRIPTION
**Pull request by WebReflection**
As mentioned in [this comment](https://github.com/whatwg/html/issues/5552#issuecomment-673181926), this PR provides a basic asynchronous test to verify the defined class is resolved.

### CODE REVIEW

**Comment by nostalic** [comment]
Hey @WebReflection, 

Could you have a look at https://github.com/web-platform-tests/wpt/pull/25030/files ?
I believe a few existing tests also needs to be updated in this PR for the new behavior.

**Review by rniwa — APPROVED** [review]
Verified the test change with the implementation change.

File: custom-elements/CustomElementRegistry.html [hunk 1]
```diff
@@ -721,17 +723,24 @@
...
+        assert_equals(promiseAfterDefine.resolved, ElementDefinedAfterWhenDefined,
+            'The promise returned by "whenDefined" must be resolved with the constructor of the element when a custom element is defined');
         assert_false('rejected' in promiseAfterDefine, 'The promise returned by "whenDefined" must not be rejected when a custom element is defined');
     });
 }, 'A promise returned by customElements.whenDefined must be resolved by "define"');
 
+promise_test(function () {
+    class ResolvedCustomElement extends HTMLElement {};
+    customElements.define('resolved-custom-element', ResolvedCustomElement);
+    return customElements.whenDefined('resolved-custom-element').then(function (value) {
```
Comment (rniwa) [inline custom-elements/CustomElementRegistry.html:740]: We should another test case where we call whenDefined before define is called.

**Comment by rniwa** [comment]
FWIW, I have a patch up on https://bugs.webkit.org/show_bug.cgi?id=215562 to implement this behavior in WebKit.

File: custom-elements/CustomElementRegistry.html (hunk 1 above)
Comment (WebReflection) [reply custom-elements/CustomElementRegistry.html:740]: Done

**Review by rniwa — APPROVED** [review]
r=me

**Pull request merged**